### PR TITLE
fix(components): [loading] loading text can be updated

### LINKED ISSUE #9263: [Component] [loading] loading 在element-loading-text 值改变后，页面不会进行对应的变化
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.12`
- Browser / OS: `chrome 103.0.5060.53`
- Build Tool: `CDN`

### Related Component

- `el-loading`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uXG4gICAgQGNsaWNrPVwiY2xpY2tcIlxuICA+XG4gICAgdGVzdFxuICA8L2VsLWJ1dHRvbj5cbiAge3ttc2d9fVxuICA8ZWwtdGFibGVcbiAgICB2LWxvYWRpbmc9XCJsb2FkaW5nXCJcbiAgICA6ZWxlbWVudC1sb2FkaW5nLXRleHQ9XCJtc2dcIlxuICAgIDplbGVtZW50LWxvYWRpbmctc3Bpbm5lcj1cInN2Z1wiXG4gICAgZWxlbWVudC1sb2FkaW5nLXN2Zy12aWV3LWJveD1cIi0xMCwgLTEwLCA1MCwgNTBcIlxuICAgIGVsZW1lbnQtbG9hZGluZy1iYWNrZ3JvdW5kPVwicmdiYSgxMjIsIDEyMiwgMTIyLCAwLjgpXCJcbiAgICA6ZGF0YT1cInRhYmxlRGF0YVwiXG4gICAgc3R5bGU9XCJ3aWR0aDogMTAwJVwiXG4gID5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJkYXRlXCIgbGFiZWw9XCJEYXRlXCIgd2lkdGg9XCIxODBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIk5hbWVcIiB3aWR0aD1cIjE4MFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiYWRkcmVzc1wiIGxhYmVsPVwiQWRkcmVzc1wiIC8+XG4gIDwvZWwtdGFibGU+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBsb2FkaW5nID0gcmVmKHRydWUpXG5jb25zdCBzdmcgPSBgXG4gICAgICAgIDxwYXRoIGNsYXNzPVwicGF0aFwiIGQ9XCJcbiAgICAgICAgICBNIDMwIDE1XG4gICAgICAgICAgTCAyOCAxN1xuICAgICAgICAgIE0gMjUuNjEgMjUuNjFcbiAgICAgICAgICBBIDE1IDE1LCAwLCAwLCAxLCAxNSAzMFxuICAgICAgICAgIEEgMTUgMTUsIDAsIDEsIDEsIDI3Ljk5IDcuNVxuICAgICAgICAgIEwgMTUgMTVcbiAgICAgICAgXCIgc3R5bGU9XCJzdHJva2Utd2lkdGg6IDRweDsgZmlsbDogcmdiYSgwLCAwLCAwLCAwKVwiLz5cbiAgICAgIGBcbmNvbnN0IHRhYmxlRGF0YSA9IFtcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAyJyxcbiAgICBuYW1lOiAnSm9obiBTbWl0aCcsXG4gICAgYWRkcmVzczogJ05vLjE1MTgsICBKaW5zaGFqaWFuZyBSb2FkLCBQdXR1byBEaXN0cmljdCcsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ0pvaG4gU21pdGgnLFxuICAgIGFkZHJlc3M6ICdOby4xNTE4LCAgSmluc2hhamlhbmcgUm9hZCwgUHV0dW8gRGlzdHJpY3QnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDEnLFxuICAgIG5hbWU6ICdKb2huIFNtaXRoJyxcbiAgICBhZGRyZXNzOiAnTm8uMTUxOCwgIEppbnNoYWppYW5nIFJvYWQsIFB1dHVvIERpc3RyaWN0JyxcbiAgfSxcbl1cblxuY29uc3QgbXNnID0gcmVmKCcxMjMnKVxuXG5cbmNvbnN0IGNsaWNrID0gKCkgPT4ge1xuICBtc2cudmFsdWUgPSAnNDU2J1xufVxuXG5cbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

点击按钮

### What is Expected?

loading text 显示 456

### What is actually happening?

依旧是123

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by baiwusanyu-c**
close: #9263  
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/loading/src/directive.ts [hunk 1]
```diff
@@ -27,16 +33,10 @@ const createInstance = (
...
-    const data = (isString(key) && vm?.[key]) || key
-    if (data) return ref(data)
-    else return data
-  }
-
   const getProp = <K extends keyof LoadingOptions>(name: K) =>
     resolveExpression(
       getBindingProp(name) ||
-        el.getAttribute(`element-loading-${hyphenate(name)}`)
+        el.getAttribute(`element-loading-${hyphenate(name)}`, vm)
```
Comment (btea) [inline packages/components/loading/src/directive.ts:39]: Why add `vm` here？

**Comment by baiwusanyu-c** [comment]
WDYT @btea

File: packages/components/loading/src/directive.ts [hunk 2]
```diff
@@ -70,6 +70,16 @@ const updateOptions = (
   }
 }
 
+const updateDirectiveAttr = (el, binding, originalOptions: LoadingOptions) => {
+  const getProp = <K extends keyof LoadingOptions>(name: K) =>
+    resolveExpression(el.getAttribute(`element-loading-${hyphenate(name)}`))
```
Comment (btea) [inline packages/components/loading/src/directive.ts:75]: This `getProp` seems to be the same function as line 36, can it be reused?

**Comment by sxzz** [comment]
https://github.com/element-plus/element-plus/issues/9263#issuecomment-1229220463

**Pull request closed without merging**